src/identity.rs

```diff
@@ -0,0 +1,844 @@
+    fn from(pod_id: PodIdentity) -> Self {
+        [
+            pod_id.app,
+            pod_id.instance,
+            pod_id.role,
+            pod_id.group,
+            pod_id.id,
+        ]
+        .join(SEMICOLON)
+    }
+}
+
+const DEFAULT_NODE_NAME: &str = "<no-nodename-set>";
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NodeIdentity {
+    pub name: String,
+}
+
+impl NodeIdentity {
+    pub fn new(name: &str) -> Self {
+        NodeIdentity {
+            name: String::from(name),
+        }
+    }
+}
+
+impl TryFrom<&Pod> for NodeIdentity {
+    type Error = Error;
+    fn try_from(p: &Pod) -> Result<Self, Error> {
+        let node_name = p
+            .spec
+            .as_ref()
+            .map(|s| s.node_name.as_ref())
+            .ok_or(Error::NodeWithoutNameNotSupported)?;
+
+        Ok(NodeIdentity {
+            name: node_name.unwrap().clone(),
+        })
+    }
+}
+
+impl From<Node> for NodeIdentity {
+    fn from(node: Node) -> Self {
+        NodeIdentity {
+            name: node
+                .metadata
+                .name
+                .unwrap_or_else(|| DEFAULT_NODE_NAME.to_string()),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PodToNodeMapping {
+    pub mapping: BTreeMap<PodIdentity, NodeIdentity>,
+}
+
+impl PodToNodeMapping {
+    /// Return a mapping for `pods` given that the `id_factory` can convert them to pod identities.
+    /// Return an `Error` if any of the given pods cannot be converted to a pod identity or
+    /// is not mapped to a node.
+    /// # Argumens
+    /// - `id_factory` : A factory that can build a `PodIdentity` from a `Pod`.
+    /// - `pods` : The pods to add to the mapping.
+    pub fn try_from(id_factory: &dyn PodIdentityFactory, pods: &[Pod]) -> Result<Self, Error> {
+        let mut mapping = BTreeMap::default();
+
+        for (pod_id, pod) in id_factory.try_from(pods)?.iter().zip(pods) {
+            mapping.insert(pod_id.clone(), NodeIdentity::try_from(pod)?);
+        }
+
+        Ok(PodToNodeMapping { mapping })
+    }
+
+    pub fn iter(&self) -> Iter<'_, PodIdentity, NodeIdentity> {
+        self.mapping.iter()
+    }
+
+    pub fn get_filtered(&self, role: &str, group: &str) -> BTreeMap<PodIdentity, NodeIdentity> {
+        let mut filtered = BTreeMap::new();
+        for (pod_id, node_id) in &self.mapping {
+            if role == pod_id.role() && pod_id.group() == group {
+                filtered.insert(pod_id.clone(), node_id.clone());
+            }
+        }
+        filtered
+    }
+
+    pub fn get(&self, pod_id: &PodIdentity) -> Option<&NodeIdentity> {
+        self.mapping.get(pod_id)
+    }
+
+    pub fn insert(&mut self, pod_id: PodIdentity, node_id: NodeIdentity) -> Option<NodeIdentity> {
+        self.mapping.insert(pod_id, node_id)
+    }
```

Comment:
Same here, this is not required for the pub field. If you plan on extending this im good with keeping it. Otherwise its just wrapper code we do not really need?